apps/src/aichat/views/bot/MovingBot.tsx

```diff
@@ -0,0 +1,70 @@
+import React, {useState} from 'react';
+import Bot, {BotState} from './Bot';
+
+function pairwise(arr: [], func: Function) {
```

Comment:
The type `[]` translates to `never[]`.  You want a generic.
```suggestion
function pairwise<T>(arr: T[], func: (a: T, b: T) => void): void {
```